Sources/NIOCore/AsyncAwaitSupport.swift

```diff
@@ -240,8 +240,25 @@ extension ChannelPipeline {
     }
 }
 
+/// An error that is thrown when the number of bytes in an AsyncSequence exceeds the limit.
+///
+/// When collecting the bytes from an AsyncSequence, there is a limit up to where the content
+/// exceeds a certain threshold beyond which the content isn't matching the expected maximum size.
+/// that will be processed. This error is generally thrown when it is discovered that there are
+/// more bytes in a sequence than what was specified as the maximum. It could be that this upTo
+/// limit should be increased, or that the sequence has unexpected content in it.
+///
+/// - maxBytes: (optional) current limit on the maximum number of bytes in the sequence
 public struct NIOTooManyBytesError: Error, Hashable {
-    public init() {}
+    let maxBytes: Int?
```

Comment:
This should be a var. no need to restrict it to a let as there are no invariants or similar that need to be verified.

Reply:
Also this should be public.

Reply:
If it's var and public, then what's a good way to prevent it from being mutated after the initial throw?

Reply:
There's no reason to prevent it being mutated. And indeed, given the `init`, you can't:

```swift
extension NIOTooManyBytesError {
   mutating func getOutOfMyWayFoolsIMutateWhatIWant(maxBytes: Int?) {
       self = .init(maxBytes: maxBytes)
   }
}

Reply:
Ok, I've updated it to be public mutable.